src/Features/Core/Portable/StackTraceExplorer/DotnetStackFrameParser.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.StackTraceExplorer
+{
+    internal sealed class DotnetStackFrameParser : IStackFrameParser
+    {
+        private const string StackTraceAtStart = "at ";
+        private const string StackTraceSymbolAndFileSplit = " in ";
+
+        /// <summary>
+        /// Tries to parse a StackFrame following convention from Environment.StackTrace
+        /// https://docs.microsoft.com/en-us/dotnet/api/system.environment.stacktrace has
+        /// details on output format and expected strings
+        /// </summary>
+        public bool TryParseLine(string line, [NotNullWhen(true)] out ParsedFrame? parsedFrame)
+        {
+            parsedFrame = null;
+
+            var atIndex = line.IndexOf(StackTraceAtStart);
+
+            if (atIndex < 0)
+            {
+                return false;
+            }
+
+            var start = atIndex + StackTraceAtStart.Length;
+
+            var endIndex = line.IndexOf(StackTraceSymbolAndFileSplit);
+            var end = endIndex > 0
+                ? endIndex
+                : line.Length;
+
+            var success = StackFrameParserHelpers.TryParseMethodSignature(line, start: start, end: end, out var classSpan, out var methodSpan, out var argsSpan);
+
+            if (!success)
+            {
+                return false;
+            }
+
+            // The line has " in <filename>:line <line number>"
+            if (endIndex > 0)
+            {
+                var fileInformationStart = endIndex + StackTraceSymbolAndFileSplit.Length;
+                var fileInformationSpan = new TextSpan(fileInformationStart, line.Length - fileInformationStart);
+
+                parsedFrame = new ParsedFrameWithFile(line, classSpan, methodSpan, argsSpan, fileInformationSpan);
+                return true;
+            }
+
+            parsedFrame = new ParsedStackFrame(line, classSpan, methodSpan, argsSpan);
```

Comment:
the names not matching bugs me :)  like ParsedFrameXXX and ParsedStackFrame.  also, consider that fileInformationSpan could just be optional in ParsedStackFrame.  that feels cleaner to me.